client/web/src/components/popovers/ColorPicker.vue

```diff
@@ -115,13 +117,197 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
+import { clamp, HSV2RGB, RGB2HSV, RGB, HSV } from "../../lib/utils";
+
+interface Rect {
+	width: number;
+	height: number;
+	left: number;
+	top: number;
+}
+
+interface ColorPickerData {
+	color: HSV;
+	hue: {
+		rect: Rect;
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
+
+type ColorPickerState = "idle" | "move_hue" | "move_opacity" | "move_saturation";
 
 export default defineComponent({
+	colorPicker: {
+		color: {
+			h: 0,
+			s: 0,
+			v: 0,
+			a: 1,
+		},
+		hue: {
+			rect: {
+				width: 0,
+				height: 0,
+				top: 0,
+				left: 0,
+			},
+		},
+		opacity: {
+			rect: {
+				width: 0,
+				height: 0,
+				top: 0,
+				left: 0,
+			},
+		},
+		saturation: {
+			rect: {
+				width: 0,
+				height: 0,
+				top: 0,
+				left: 0,
+			},
+		},
```

Comment:
Please format in a more condensed format to make these low-entropy lines take up less space:

```ts
		color: { h: 0, s: 0, v: 0, a: 1 },
		hue: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
		opacity: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
		saturation: {
			rect: { width: 0, height: 0, top: 0, left: 0 },
		},
```